mmedit/datasets/pipelines/crop.py

```diff
@@ -39,15 +45,25 @@ def _crop(self, data):
         for item in data_list:
             data_h, data_w = item.shape[:2]
             crop_h, crop_w = self.crop_size
-            crop_h = min(data_h, crop_h)
-            crop_w = min(data_w, crop_w)
 
             if self.random_crop:
+                crop_h = min(data_h, crop_h)
+                crop_w = min(data_w, crop_w)
                 x_offset = np.random.randint(0, data_w - crop_w + 1)
                 y_offset = np.random.randint(0, data_h - crop_h + 1)
             else:
-                x_offset = max(0, (data_w - crop_w)) // 2
-                y_offset = max(0, (data_h - crop_h)) // 2
+
+                x_offset = abs(data_w - crop_w) // 2
+                y_offset = abs(data_h - crop_h) // 2
+
+                if crop_h > data_h:
+                    pad_width = ((2 * y_offset, 2 * y_offset), (0, 0), (0, 0))
+                    data = np.pad(
+                        data, pad_width, mode='constant', constant_values=0)
+                elif crop_w > data_w:
+                    pad_width = ((0, 0), (2 * x_offset, 2 * x_offset), (0, 0))
+                    data = np.pad(
+                        data, pad_width, mode='constant', constant_values=0)
```

Comment:
This part can be moved to line 48. After padding, update `crop_h, crop_w, data_h, data_w`.

In this case, the original code can be kept unchanged.